[Stack.Item] Add Stack.Item properties to style guide documentation (#772)
Shopify/polaris-react: src/components/Stack/README.md

---

## Stack item

> Use item to align buttons or secondary content to the right edge of another element, allowing it to wrap below on small screens.

Reviewer: The `Stack` automatically wraps each of its children in a `Stack.Item` (unless the implementor has already wrapped a child/children with `Stack.Item`). It may be helpful to mention that, and then explain that you can use the subcomponent directly with the `fill` prop when you want the item to fill the available horizontal space in the stack (and mention they can see this in action in the "Stack where a single item fills the remaining space" example above).

> The Stack will implicitly wrap each of its children in a Stack.Item unless used directly. Use Stack.Item as a subcomponent with the fill prop to have the item fill the available horizontal space in the stack. See "Stack where a single item fills the remaining space" as an example of this above.

Reviewer 2: The Stack will implicitly wrap each of its children in a Stack.Item unless used directly 👍 I'm glad we've finally gotten this documented ❤️

Reviewer 3: Yes! I'm also happy to see this get documented!

I don't think we use determiners before component names, unless it's followed by the word "component". So, something like `The stack component will implicitly...`.

In documentation, we also use sentence case for components, ie: a stack item vs. Stack.Item. https://docs.google.com/document/d/1zVDsHIWhoir2svRjdtSdRbD_ruTz3K1nAJcQLGPVQwM/edit#heading=h.lm3ja0wzryt7

It might be a good idea to link to the example using a named anchor. The resource list documentation has an example of this.

### Stack item properties

| Prop     | Type    | Description                                                                | Default |
| -------- | ------- | -------------------------------------------------------------------------- | ------- |

> | fill     | boolean | Adjust vertical alignment of elements to the right edge of another element | false   |

Reviewer: ```suggestion
 | fill | boolean | Fill the available horizontal space in the stack with the item | false |
```

> | fill     | boolean | Fil the available horizontal space in the stack with the item | false   |

Reviewer 2: Would it be worth it to add these to the interface as well?

---

## Stack item

> The stack component will wrap each item in a stack item component unless an item has already been deliberately wrapped in a stack item component. Use the stack item component with the fill prop to force the item to fill the remaining horizontal space in the stack. See the [Stack where a single item fills the remaining space](#single-item-fills-remaining-space) example.

Author: @selenehinkley Would you be able to take a look at the  current wording of this and let me know what you think?

Reviewer 4: 🎉 Stack will treat multiple elements wrapped in a stack item component as one item. By default, each individual element is treated as one item. Use the fill prop on a single stack item to make it fill the rest of the available horizontal space.

> The stack component will wrap each item in a stack item component unless an item has already been deliberately wrapped in a stack item component. Use the stack item component with the fill prop to force the item to fill the remaining horizontal space in the stack. See the [Stack where a single item fills the remaining space](#single-item-fills-remaining-space) example.

Reviewer 4: ```suggestion
The stack component wraps all items in the stack item component unless one item has already been deliberately wrapped. Use the stack item component with the fill prop to force the item to fill the remaining horizontal space in the stack. See the [Stack where a single item fills the remaining space](#single-item-fills-remaining-space) example.
```

> The stack component will treat multiple elements wrapped in a stack item component as one item. By default, each individual element is treated as one stack item. Use the fill prop on a single stack item component to make it fill the rest of the available horizontal space. See the [Stack where a single item fills the remaining space](#single-item-fills-remaining-space) example.

Reviewer 3: Perfect!! ❤️

> The stack component will wrap each item in a stack item component regardless if the item has already been deliberately wrapped in a stack item component or not. However, When wrapping the stack component with a stack item deliberately, use the fill prop to force the item to fill the remaining horizontal space in the stack. See the [Stack where a single item fills the remaining space](#single-item-fills-remaining-space) example.

Reviewer 3: `The stack component will wrap each item in a stack item component regardless if the item has already been deliberately wrapped in a stack item component or not.`

To me that sounds like an item could potentially be wrapped twice in a stack item component. The word "regardless" sounds like it ignores what it's been given and goes ahead and wraps another stack item around it.

When what it actually does is wrap each item in a stack item component *if it hasn't* been deliberately wrapped already...

`However, When wrapping the stack component with a stack item deliberately, use the fill prop...`

This sounds like it's mandatory. To me, this reads like "if you decide to wrap it deliberately, you have to use the fill prop". Maybe something like `you can use the fill prop...`.
